File: algorithms/1_easy/E/04020_elevator_requests_i/solution.c
int elevatorRequests(const int n, const int *restrict requests, const int len) {
    int prev = 0;
    int res = 0;

    for (int i = 0; i < len; i += 1) {
        res += ABS(prev - requests[i]);
        prev = requests[i];
    }

    return res;
}

File: algorithms/1_easy/E/04020_elevator_requests_i/solution.rs
impl Solution {
    pub fn elevator_requests(n: i32, requests: Vec<i32>) -> i32 {
        let mut prev = 0_i32;
        let mut res = 0_i32;

        for r in requests {
            res += (prev - r).abs();
            prev = r;
        }

        res
    }
}
